homeassistant/components/starlink/coordinator.py

```diff
@@ -0,0 +1,61 @@
+"""Contains the shared Coordinator for Starlink systems."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from starlink_grpc import ChannelContext, GrpcError, get_status
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .dish_status import DishyStatus
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class StarlinkUpdateCoordinator(DataUpdateCoordinator[DishyStatus]):
+    """Coordinates updates between all Starlink sensors defined in this file."""
+
+    def __init__(self, hass: HomeAssistant, name: str, url: str) -> None:
+        """Initialize an UpdateCoordinator for a group of sensors."""
+        self.channel_context = ChannelContext(target=url)
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=name,
+            update_interval=timedelta(seconds=1),
+            request_refresh_debouncer=Debouncer(
```

Comment:
This would probably not be needed with longer update intervals

Reply:
The debouncer probably isn't enev needed now, it was implemented to work around a flaky API Buti since changed how the integration fetches data. I'll remove it